pkg/controller/controller.go

```diff
@@ -166,56 +181,87 @@ func (haproxy *HAProxyController) OnUpdate(cfg ingress.Configuration) error {
 		return nil
 	}
 
-	out, err := haproxy.reloadHaproxy()
+	reloadCmd := haproxy.reloadHaproxy(configFile)
+	out, err := reloadCmd.CombinedOutput()
 	if len(out) > 0 {
-		glog.Infof("HAProxy output:\n%v", string(out))
+		glog.Infof("HAProxy[pid=%v] output:\n%v", reloadCmd.Process.Pid, string(out))
 	}
 	return err
 }
 
 // RewriteConfigFiles safely replaces configuration files with new contents after validation
-func rewriteConfigFiles(data []byte, reloadStrategy, configFile string) error {
-	tmpf := "/etc/haproxy/new_cfg.erb"
-
-	err := ioutil.WriteFile(tmpf, data, 644)
-	if err != nil {
-		glog.Warningln("Error writing rendered template to file")
-		return err
+func (haproxy *HAProxyController) rewriteConfigFiles(data []byte) (string, error) {
+	// Include timestamp in config file name to aid troubleshooting. When using a single, ever-changing config file it
+	// was difficult to know what config was loaded by any given haproxy process
+	timestamp := time.Now().Format("-060102-150405.0000")
+	if *haproxy.reloadStrategy == "multibinder" {
+		// multibinder currently limited to fixed config file path
+		timestamp = ""
 	}
-
-	if reloadStrategy == "multibinder" {
-		generated, err := multibinderERBOnly(tmpf)
-		if err != nil {
-			return err
+	configFile := haproxy.configDir + "/" + haproxy.configFilePrefix + timestamp + haproxy.configFileSuffix
+
+	if *haproxy.reloadStrategy == "multibinder" {
+		erbFile := configFile + ".erb"
+		// Write to ERB template file
+		if err := ioutil.WriteFile(erbFile, data, 644); err != nil {
+			glog.Warningln("Error writing rendered template to file")
+			return "", err
 		}
-		err = os.Rename(generated, "/etc/haproxy/haproxy.cfg")
-		if err != nil {
-			glog.Warningln("Error updating config file")
-			return err
+
+		// Generate configFile contents by processing ERB template (also validates haproxy config)
+		if err := multibinderERBOnly(erbFile); err != nil {
+			return "", err
 		}
 	} else {
-		err = checkValidity(tmpf)
-		if err != nil {
-			return err
+		// Write directly to configFile
+		if err := ioutil.WriteFile(configFile, data, 644); err != nil {
+			glog.Warningln("Error writing rendered template to file")
+			return "", err
+		}
+
+		// Validate haproxy config
+		if err := checkValidity(configFile); err != nil {
+			return "", err
 		}
 	}
-	err = os.Rename(tmpf, configFile)
+
+	haproxy.removeOldConfigFiles(haproxy.configFileSuffix)
+	return configFile, nil
+}
+
+func (haproxy *HAProxyController) removeOldConfigFiles(suffix string) error {
+	files, err := ioutil.ReadDir(haproxy.configDir)
 	if err != nil {
-		glog.Warningln("Error updating config file")
 		return err
 	}
```

Comment:
error isn't being used, so log a warningf("...: %v", err) here and return without an argument